### PR TITLE
Allow comments in the forward-zones-file

### LINKED ISSUE #2430: Please allow a comment character in the forward-zones-file
Please allow a comment character (And empty lines) in the forward-zones-file.

Example:

```
# AOL
aol.com=127.0.0.1

# Microsoft
microsoft.com=192.168.1.1
zone.microsoft.com=192.168.2.1
```

TIA,

### PR DESCRIPTION
**Pull request by pieterlexis**
* Closes #2430
  * Lines starting with a '#' get skipped.
  * Everything after a '#' symbol on a line is ignored
  * Document this change

### CODE REVIEW

File: docs/markdown/recursor/settings.md [hunk 1]
```diff
@@ -272,12 +272,13 @@ setting is intended to forward queries to authoritative servers.
 
 Same as [`forward-zones`](#forward-zones), parsed from a file. Only 1 zone is
 allowed per line, specified as follows: `example.org=203.0.113.210, 192.0.2.4:5300`.
-No comments are allowed.
 
 Since version 3.2, zones prefixed with a '+' are forwarded with the
 recursion-desired bit set to one, for which see ['forward-zones-recurse'](#forward-zones-recurse).
 Default behaviour without '+' is as with [`forward-zones`](#forward-zones).
 
+Shell-style comments using '#' are allowed since version 4.0.
```
Comment (Habbie) [inline docs/markdown/recursor/settings.md]: 4.0.0 please :)

**Comment by Habbie** [comment]
LGTM. How about a test? :)

**Comment by pieterlexis** [comment]
@Habbie hows this?

**Comment by pieterlexis** [comment]
comments:

* As everything after a '#' is ignored, it's not shell, document this
* Test is horrible, this info can be retrieved from the API, PR will be finished after `rec_control` uses the API.

**Comment by pieterlexis** [comment]
@Habbie ready to merge!

**Comment by Habbie** [comment]
"As everything after a '#' is ignored, it's not shell, document this"

this does not look documented yet. If you fix that nit, ready for merge.

**Comment by Habbie** [comment]
Withdrawn!

**Pull request merged**